core/src/avm2/vtable.rs

```diff
@@ -473,6 +474,19 @@ impl<'gc> VTable<'gc> {
             write.resolved_traits.insert(interface_name, prop);
         }
     }
+
+    pub fn public_properties(self) -> Vec<(AvmString<'gc>, Property)> {
+        let read = self.0.read();
+
+        let mut props = Vec::new();
+
+        for (name, ns, prop) in read.resolved_traits.iter() {
+            if ns.is_public() {
+                props.push((name, *prop));
+            }
+        }
+        props
+    }
```

Comment:
Instead of creating a new vector here, maybe we could return an iterator that only yields public properties

Reply:
That's tricky, because we we need to access them through the `Ref`

Reply:
Hm yeah, I thought it might be possible through `Ref::map`, but it doesn't seem like it